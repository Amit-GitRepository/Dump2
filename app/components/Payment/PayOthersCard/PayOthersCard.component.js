import Button from '../../Shared/Button/Button.component';
import IconTextInput from '../../Shared/IconTextInput/IconTextInput.component';
import Proptypes from 'prop-types';
import React, {Component} from 'react';
import {ISText} from '../../Shared/index';
import {Keyboard, View} from 'react-native';
import {styles} from './PayOthersCard.style';
import {translate} from '../../../language/i18n/helper';
 
class PayOthersCard extends Component {

  state=  {
    buttonDisabled: true
  }

  setServiceNo = (newServiceNo) => this.serviceNo = newServiceNo;

  navigate = () => {
    Keyboard.dismiss();
    if (this.serviceNo)
      this.props.fetchPaymentProducts(this.serviceNo); 
  }

  onSubmitEditing = () => {
    this.navigate();
  }

  onTextInputChange = (text) => {
    const {buttonDisabled} = this.state;
    if (text && buttonDisabled) {
      this.setState({buttonDisabled: false});
    } else if (!text && !buttonDisabled) {
      this.setState({buttonDisabled: true});
    }
    this.serviceNo = text;
  }

  render () {
    const {buttonDisabled} = this.state;
    return (
      <View style={styles.container}>
        <View style={styles.card}>
          <ISText style={styles.headerText} type='SEMIBOLD'>{translate('PAYMENT_OTHERS_ENTER_PROMPT')}</ISText>
          <IconTextInput keyboardType={'numeric'} containerStyle={styles.inputBox} placeholder={translate('PAYMENT_OTHERS_PLACEHOLDER')} onChangeText={this.onTextInputChange} onSubmitEditing={this.onSubmitEditing} iconName={'question'} iconSize={20} numberOfLines={1}/>
          <Button style={styles.button} disabled={buttonDisabled} text={translate('PAYMENT_OTHERS__NEXT')} onPress={this.navigate}/>
        </View>
      </View>
    );
  }
}
PayOthersCard.propTypes = {
  fetchPaymentProducts: Proptypes.func
};
export default PayOthersCard;