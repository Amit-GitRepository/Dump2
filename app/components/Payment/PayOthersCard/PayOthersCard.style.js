import {colors, fonts} from '../../../themes/constants.styles';
import {shadowActiveStyle} from '../../../themes/application.styles';
 
export const styles = {
  card: {
    justifyContent: 'space-between',
    backgroundColor: colors.PRIMARY_BG_TEXT_CONTRAST,
    padding: 15,
    borderRadius: 8,
    ...shadowActiveStyle
  },
  container: {
    padding: 10
  },
  headerText: {
    fontSize: fonts.FONT_SIZE_NORMAL,
    marginBottom: 10
  },
  inputBox: {
    borderWidth: 1,
    borderRadius: 8,
    borderColor: colors.PRIMARY_DISABLED_BG_TEXT,
    paddingHorizontal: 5,
    flexDirection: 'row-reverse'
  },
  button: {
    alignSelf: 'center',
    marginTop: 10,
    width: 100
  }
};