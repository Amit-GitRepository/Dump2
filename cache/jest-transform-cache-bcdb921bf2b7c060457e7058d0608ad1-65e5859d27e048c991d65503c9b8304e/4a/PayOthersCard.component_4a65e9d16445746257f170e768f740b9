Object.defineProperty(exports,"__esModule",{value:true});var cov_1g7e30prk6=function(){var path='/Users/amitkumar/iTrueService/DevCode/iservice-app/app/components/Payment/PayOthersCard/PayOthersCard.component.js',hash='069bf9b27a7d9acaf364ecb13034884400ac3233',global=new Function('return this')(),gcv='__coverage__',coverageData={path:'/Users/amitkumar/iTrueService/DevCode/iservice-app/app/components/Payment/PayOthersCard/PayOthersCard.component.js',statementMap:{'0':{start:{line:12,column:35},end:{line:12,column:64}},'1':{start:{line:15,column:4},end:{line:15,column:23}},'2':{start:{line:16,column:4},end:{line:17,column:54}},'3':{start:{line:17,column:6},end:{line:17,column:54}},'4':{start:{line:21,column:4},end:{line:21,column:26}},'5':{start:{line:25,column:4},end:{line:33,column:6}},'6':{start:{line:36,column:0},end:{line:38,column:2}}},fnMap:{'0':{name:'(anonymous_0)',decl:{start:{line:12,column:17},end:{line:12,column:18}},loc:{start:{line:12,column:35},end:{line:12,column:64}},line:12},'1':{name:'(anonymous_1)',decl:{start:{line:14,column:13},end:{line:14,column:14}},loc:{start:{line:14,column:19},end:{line:18,column:3}},line:14},'2':{name:'(anonymous_2)',decl:{start:{line:20,column:22},end:{line:20,column:23}},loc:{start:{line:20,column:32},end:{line:22,column:3}},line:20},'3':{name:'(anonymous_3)',decl:{start:{line:24,column:2},end:{line:24,column:3}},loc:{start:{line:24,column:12},end:{line:34,column:3}},line:24}},branchMap:{'0':{loc:{start:{line:16,column:4},end:{line:17,column:54}},type:'if',locations:[{start:{line:16,column:4},end:{line:17,column:54}},{start:{line:16,column:4},end:{line:17,column:54}}],line:16}},s:{'0':0,'1':0,'2':0,'3':0,'4':0,'5':0,'6':0},f:{'0':0,'1':0,'2':0,'3':0},b:{'0':[0,0]},_coverageSchema:'332fd63041d2c1bcb487cc26dd0d5f7d97098a6c'},coverage=global[gcv]||(global[gcv]={});if(coverage[path]&&coverage[path].hash===hash){return coverage[path];}coverageData.hash=hash;return coverage[path]=coverageData;}();var _createClass=function(){function defineProperties(target,props){for(var i=0;i<props.length;i++){var descriptor=props[i];descriptor.enumerable=descriptor.enumerable||false;descriptor.configurable=true;if("value"in descriptor)descriptor.writable=true;Object.defineProperty(target,descriptor.key,descriptor);}}return function(Constructor,protoProps,staticProps){if(protoProps)defineProperties(Constructor.prototype,protoProps);if(staticProps)defineProperties(Constructor,staticProps);return Constructor;};}();var _Button=require('../../Shared/Button/Button.component');var _Button2=_interopRequireDefault(_Button);
var _IconTextInput=require('../../Shared/IconTextInput/IconTextInput.component');var _IconTextInput2=_interopRequireDefault(_IconTextInput);
var _propTypes=require('prop-types');var _propTypes2=_interopRequireDefault(_propTypes);
var _react=require('react');var _react2=_interopRequireDefault(_react);
var _index=require('../../Shared/index');
var _reactNative=require('react-native');
var _PayOthersCard=require('./PayOthersCard.style');
var _helper=require('../../../language/i18n/helper');function _interopRequireDefault(obj){return obj&&obj.__esModule?obj:{default:obj};}function _classCallCheck(instance,Constructor){if(!(instance instanceof Constructor)){throw new TypeError("Cannot call a class as a function");}}function _possibleConstructorReturn(self,call){if(!self){throw new ReferenceError("this hasn't been initialised - super() hasn't been called");}return call&&(typeof call==="object"||typeof call==="function")?call:self;}function _inherits(subClass,superClass){if(typeof superClass!=="function"&&superClass!==null){throw new TypeError("Super expression must either be null or a function, not "+typeof superClass);}subClass.prototype=Object.create(superClass&&superClass.prototype,{constructor:{value:subClass,enumerable:false,writable:true,configurable:true}});if(superClass)Object.setPrototypeOf?Object.setPrototypeOf(subClass,superClass):subClass.__proto__=superClass;}var

PayOthersCard=function(_Component){_inherits(PayOthersCard,_Component);function PayOthersCard(){var _ref;var _temp,_this,_ret;_classCallCheck(this,PayOthersCard);for(var _len=arguments.length,args=Array(_len),_key=0;_key<_len;_key++){args[_key]=arguments[_key];}return _ret=(_temp=(_this=_possibleConstructorReturn(this,(_ref=PayOthersCard.__proto__||Object.getPrototypeOf(PayOthersCard)).call.apply(_ref,[this].concat(args))),_this),_this.

setServiceNo=function(newServiceNo){cov_1g7e30prk6.f[0]++;cov_1g7e30prk6.s[0]++;return _this.serviceNo=newServiceNo;},_this.

navigate=function(){cov_1g7e30prk6.f[1]++;cov_1g7e30prk6.s[1]++;
_reactNative.Keyboard.dismiss();cov_1g7e30prk6.s[2]++;
if(_this.serviceNo)
{cov_1g7e30prk6.b[0][0]++;cov_1g7e30prk6.s[3]++;_this.props.fetchPaymentProducts(_this.serviceNo);}else{cov_1g7e30prk6.b[0][1]++;}
},_this.

onTextInputChange=function(text){cov_1g7e30prk6.f[2]++;cov_1g7e30prk6.s[4]++;
_this.serviceNo=text;
},_temp),_possibleConstructorReturn(_this,_ret);}_createClass(PayOthersCard,[{key:'render',value:function render()

{cov_1g7e30prk6.f[3]++;cov_1g7e30prk6.s[5]++;
return(
_react2.default.createElement(_reactNative.View,{style:_PayOthersCard.styles.container},
_react2.default.createElement(_reactNative.View,{style:_PayOthersCard.styles.card},
_react2.default.createElement(_index.ISText,{style:_PayOthersCard.styles.headerText,type:'SEMIBOLD'},(0,_helper.translate)('PAYMENT_OTHERS_ENTER_PROMPT')),
_react2.default.createElement(_IconTextInput2.default,{containerStyle:_PayOthersCard.styles.inputBox,placeholder:(0,_helper.translate)('PAYMENT_OTHERS_PLACEHOLDER'),onChangeText:this.onTextInputChange}),
_react2.default.createElement(_Button2.default,{style:_PayOthersCard.styles.button,text:'Next',onPress:this.navigate}))));



}}]);return PayOthersCard;}(_react.Component);cov_1g7e30prk6.s[6]++;

PayOthersCard.propTypes={
fetchPaymentProducts:_propTypes2.default.func};exports.default=

PayOthersCard;