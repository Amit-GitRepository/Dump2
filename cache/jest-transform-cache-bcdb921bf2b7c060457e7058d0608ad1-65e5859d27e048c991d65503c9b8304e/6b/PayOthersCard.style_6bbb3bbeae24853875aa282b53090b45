Object.defineProperty(exports,"__esModule",{value:true});exports.styles=undefined;var cov_26rppu4kpj=function(){var path='/Users/amitkumar/iTrueService/DevCode/iservice-app/app/components/Payment/PayOthersCard/PayOthersCard.style.js',hash='d8a5c9ee08fb356429cadb6077efd1754b11974e',global=new Function('return this')(),gcv='__coverage__',coverageData={path:'/Users/amitkumar/iTrueService/DevCode/iservice-app/app/components/Payment/PayOthersCard/PayOthersCard.style.js',statementMap:{'0':{start:{line:4,column:22},end:{line:27,column:1}}},fnMap:{},branchMap:{},s:{'0':0},f:{},b:{},_coverageSchema:'332fd63041d2c1bcb487cc26dd0d5f7d97098a6c'},coverage=global[gcv]||(global[gcv]={});if(coverage[path]&&coverage[path].hash===hash){return coverage[path];}coverageData.hash=hash;return coverage[path]=coverageData;}();var _extends=Object.assign||function(target){for(var i=1;i<arguments.length;i++){var source=arguments[i];for(var key in source){if(Object.prototype.hasOwnProperty.call(source,key)){target[key]=source[key];}}}return target;};var _constants=require('../../../themes/constants.styles');
var _application=require('../../../themes/application.styles');

var styles=exports.styles=(cov_26rppu4kpj.s[0]++,{
card:_extends({
justifyContent:'space-between',
backgroundColor:_constants.colors.PRIMARY_BG_TEXT_CONTRAST,
padding:15,
borderRadius:8},_application.shadowActiveStyle),


container:{
padding:10},

headerText:{
fontSize:_constants.fonts.FONT_SIZE_NORMAL,
marginBottom:10},

inputBox:{
borderWidth:1},

button:{
alignSelf:'center',
marginTop:10,
width:100}});