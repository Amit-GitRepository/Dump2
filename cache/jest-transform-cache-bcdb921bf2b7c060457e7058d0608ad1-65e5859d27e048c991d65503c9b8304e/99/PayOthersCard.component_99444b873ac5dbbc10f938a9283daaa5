Object.defineProperty(exports,"__esModule",{value:true});var cov_1g7e30prk6=function(){var path='/Users/amitkumar/iTrueService/DevCode/iservice-app/app/components/Payment/PayOthersCard/PayOthersCard.component.js',hash='d21831dd3a8ae3fa1ce4ca0023821c886a3ac24d',global=new Function('return this')(),gcv='__coverage__',coverageData={path:'/Users/amitkumar/iTrueService/DevCode/iservice-app/app/components/Payment/PayOthersCard/PayOthersCard.component.js',statementMap:{'0':{start:{line:16,column:35},end:{line:16,column:64}},'1':{start:{line:19,column:4},end:{line:19,column:23}},'2':{start:{line:20,column:4},end:{line:21,column:54}},'3':{start:{line:21,column:6},end:{line:21,column:54}},'4':{start:{line:25,column:29},end:{line:25,column:39}},'5':{start:{line:26,column:4},end:{line:30,column:5}},'6':{start:{line:27,column:6},end:{line:27,column:45}},'7':{start:{line:28,column:11},end:{line:30,column:5}},'8':{start:{line:29,column:6},end:{line:29,column:44}},'9':{start:{line:31,column:4},end:{line:31,column:26}},'10':{start:{line:35,column:29},end:{line:35,column:39}},'11':{start:{line:36,column:4},end:{line:44,column:6}},'12':{start:{line:47,column:0},end:{line:49,column:2}}},fnMap:{'0':{name:'(anonymous_0)',decl:{start:{line:16,column:17},end:{line:16,column:18}},loc:{start:{line:16,column:35},end:{line:16,column:64}},line:16},'1':{name:'(anonymous_1)',decl:{start:{line:18,column:13},end:{line:18,column:14}},loc:{start:{line:18,column:19},end:{line:22,column:3}},line:18},'2':{name:'(anonymous_2)',decl:{start:{line:24,column:22},end:{line:24,column:23}},loc:{start:{line:24,column:32},end:{line:32,column:3}},line:24},'3':{name:'(anonymous_3)',decl:{start:{line:34,column:2},end:{line:34,column:3}},loc:{start:{line:34,column:12},end:{line:45,column:3}},line:34}},branchMap:{'0':{loc:{start:{line:20,column:4},end:{line:21,column:54}},type:'if',locations:[{start:{line:20,column:4},end:{line:21,column:54}},{start:{line:20,column:4},end:{line:21,column:54}}],line:20},'1':{loc:{start:{line:26,column:4},end:{line:30,column:5}},type:'if',locations:[{start:{line:26,column:4},end:{line:30,column:5}},{start:{line:26,column:4},end:{line:30,column:5}}],line:26},'2':{loc:{start:{line:26,column:8},end:{line:26,column:30}},type:'binary-expr',locations:[{start:{line:26,column:8},end:{line:26,column:12}},{start:{line:26,column:16},end:{line:26,column:30}}],line:26},'3':{loc:{start:{line:28,column:11},end:{line:30,column:5}},type:'if',locations:[{start:{line:28,column:11},end:{line:30,column:5}},{start:{line:28,column:11},end:{line:30,column:5}}],line:28},'4':{loc:{start:{line:28,column:15},end:{line:28,column:39}},type:'binary-expr',locations:[{start:{line:28,column:15},end:{line:28,column:20}},{start:{line:28,column:24},end:{line:28,column:39}}],line:28}},s:{'0':0,'1':0,'2':0,'3':0,'4':0,'5':0,'6':0,'7':0,'8':0,'9':0,'10':0,'11':0,'12':0},f:{'0':0,'1':0,'2':0,'3':0},b:{'0':[0,0],'1':[0,0],'2':[0,0],'3':[0,0],'4':[0,0]},_coverageSchema:'332fd63041d2c1bcb487cc26dd0d5f7d97098a6c'},coverage=global[gcv]||(global[gcv]={});if(coverage[path]&&coverage[path].hash===hash){return coverage[path];}coverageData.hash=hash;return coverage[path]=coverageData;}();var _createClass=function(){function defineProperties(target,props){for(var i=0;i<props.length;i++){var descriptor=props[i];descriptor.enumerable=descriptor.enumerable||false;descriptor.configurable=true;if("value"in descriptor)descriptor.writable=true;Object.defineProperty(target,descriptor.key,descriptor);}}return function(Constructor,protoProps,staticProps){if(protoProps)defineProperties(Constructor.prototype,protoProps);if(staticProps)defineProperties(Constructor,staticProps);return Constructor;};}();var _Button=require('../../Shared/Button/Button.component');var _Button2=_interopRequireDefault(_Button);
var _IconTextInput=require('../../Shared/IconTextInput/IconTextInput.component');var _IconTextInput2=_interopRequireDefault(_IconTextInput);
var _propTypes=require('prop-types');var _propTypes2=_interopRequireDefault(_propTypes);
var _react=require('react');var _react2=_interopRequireDefault(_react);
var _index=require('../../Shared/index');
var _reactNative=require('react-native');
var _PayOthersCard=require('./PayOthersCard.style');
var _helper=require('../../../language/i18n/helper');function _interopRequireDefault(obj){return obj&&obj.__esModule?obj:{default:obj};}function _classCallCheck(instance,Constructor){if(!(instance instanceof Constructor)){throw new TypeError("Cannot call a class as a function");}}function _possibleConstructorReturn(self,call){if(!self){throw new ReferenceError("this hasn't been initialised - super() hasn't been called");}return call&&(typeof call==="object"||typeof call==="function")?call:self;}function _inherits(subClass,superClass){if(typeof superClass!=="function"&&superClass!==null){throw new TypeError("Super expression must either be null or a function, not "+typeof superClass);}subClass.prototype=Object.create(superClass&&superClass.prototype,{constructor:{value:subClass,enumerable:false,writable:true,configurable:true}});if(superClass)Object.setPrototypeOf?Object.setPrototypeOf(subClass,superClass):subClass.__proto__=superClass;}var

PayOthersCard=function(_Component){_inherits(PayOthersCard,_Component);function PayOthersCard(){var _ref;var _temp,_this,_ret;_classCallCheck(this,PayOthersCard);for(var _len=arguments.length,args=Array(_len),_key=0;_key<_len;_key++){args[_key]=arguments[_key];}return _ret=(_temp=(_this=_possibleConstructorReturn(this,(_ref=PayOthersCard.__proto__||Object.getPrototypeOf(PayOthersCard)).call.apply(_ref,[this].concat(args))),_this),_this.

state={
buttonDisabled:true},_this.


setServiceNo=function(newServiceNo){cov_1g7e30prk6.f[0]++;cov_1g7e30prk6.s[0]++;return _this.serviceNo=newServiceNo;},_this.

navigate=function(){cov_1g7e30prk6.f[1]++;cov_1g7e30prk6.s[1]++;
_reactNative.Keyboard.dismiss();cov_1g7e30prk6.s[2]++;
if(_this.serviceNo)
{cov_1g7e30prk6.b[0][0]++;cov_1g7e30prk6.s[3]++;_this.props.fetchPaymentProducts(_this.serviceNo);}else{cov_1g7e30prk6.b[0][1]++;}
},_this.

onTextInputChange=function(text){cov_1g7e30prk6.f[2]++;var _ref2=(cov_1g7e30prk6.s[4]++,
_this.state),buttonDisabled=_ref2.buttonDisabled;cov_1g7e30prk6.s[5]++;
if((cov_1g7e30prk6.b[2][0]++,text)&&(cov_1g7e30prk6.b[2][1]++,buttonDisabled)){cov_1g7e30prk6.b[1][0]++;cov_1g7e30prk6.s[6]++;
_this.setState({buttonDisabled:false});
}else{cov_1g7e30prk6.b[1][1]++;cov_1g7e30prk6.s[7]++;if((cov_1g7e30prk6.b[4][0]++,!text)&&(cov_1g7e30prk6.b[4][1]++,!buttonDisabled)){cov_1g7e30prk6.b[3][0]++;cov_1g7e30prk6.s[8]++;
_this.setState({buttonDisabled:true});
}else{cov_1g7e30prk6.b[3][1]++;}}cov_1g7e30prk6.s[9]++;
_this.serviceNo=text;
},_temp),_possibleConstructorReturn(_this,_ret);}_createClass(PayOthersCard,[{key:'render',value:function render()

{cov_1g7e30prk6.f[3]++;var _ref3=(cov_1g7e30prk6.s[10]++,
this.state),buttonDisabled=_ref3.buttonDisabled;cov_1g7e30prk6.s[11]++;
return(
_react2.default.createElement(_reactNative.View,{style:_PayOthersCard.styles.container},
_react2.default.createElement(_reactNative.View,{style:_PayOthersCard.styles.card},
_react2.default.createElement(_index.ISText,{style:_PayOthersCard.styles.headerText,type:'SEMIBOLD'},(0,_helper.translate)('PAYMENT_OTHERS_ENTER_PROMPT')),
_react2.default.createElement(_IconTextInput2.default,{containerStyle:_PayOthersCard.styles.inputBox,placeholder:(0,_helper.translate)('PAYMENT_OTHERS_PLACEHOLDER'),onChangeText:this.onTextInputChange}),
_react2.default.createElement(_Button2.default,{style:_PayOthersCard.styles.button,disabled:buttonDisabled,text:'Next',onPress:this.navigate}))));



}}]);return PayOthersCard;}(_react.Component);cov_1g7e30prk6.s[12]++;

PayOthersCard.propTypes={
fetchPaymentProducts:_propTypes2.default.func};exports.default=

PayOthersCard;