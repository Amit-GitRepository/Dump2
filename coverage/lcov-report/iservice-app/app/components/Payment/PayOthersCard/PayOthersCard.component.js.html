<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for iservice-app/app/components/Payment/PayOthersCard/PayOthersCard.component.js</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../../../../prettify.css" />
    <link rel="stylesheet" href="../../../../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../../../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../../../../index.html">All files</a> / <a href="index.html">iservice-app/app/components/Payment/PayOthersCard</a> PayOthersCard.component.js
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">23.08% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>3/13</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">0% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>0/10</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">25% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>1/4</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">23.08% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>3/13</span>
      </div>
    </div>
  </div>
  <div class='status-line low'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31
32
33
34
35
36
37
38
39
40
41
42
43
44
45
46
47
48
49
50</td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">38x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">import Button from '../../Shared/Button/Button.component';
import IconTextInput from '../../Shared/IconTextInput/IconTextInput.component';
import Proptypes from 'prop-types';
import React, {Component} from 'react';
import {ISText} from '../../Shared/index';
import {Keyboard, View} from 'react-native';
import {styles} from './PayOthersCard.style';
import {translate} from '../../../language/i18n/helper';
 
class PayOthersCard extends Component {
&nbsp;
  state=  {
    buttonDisabled: true
  }
&nbsp;
  setServiceNo = <span class="fstat-no" title="function not covered" >(n</span>ewServiceNo) =&gt; <span class="cstat-no" title="statement not covered" >this.serviceNo = newServiceNo;</span>
&nbsp;
  navigate = <span class="fstat-no" title="function not covered" >()</span> =&gt; {
<span class="cstat-no" title="statement not covered" >    Keyboard.dismiss();</span>
<span class="cstat-no" title="statement not covered" >    if (this.serviceNo)</span>
<span class="cstat-no" title="statement not covered" >      this.props.fetchPaymentProducts(this.serviceNo); </span>
  }
&nbsp;
  onTextInputChange = <span class="fstat-no" title="function not covered" >(t</span>ext) =&gt; {
    const {buttonDisabled} = <span class="cstat-no" title="statement not covered" >this.state;</span>
<span class="cstat-no" title="statement not covered" >    if (text &amp;&amp; buttonDisabled) {</span>
<span class="cstat-no" title="statement not covered" >      this.setState({buttonDisabled: false});</span>
    } else <span class="cstat-no" title="statement not covered" >if (!text &amp;&amp; !buttonDisabled) {</span>
<span class="cstat-no" title="statement not covered" >      this.setState({buttonDisabled: true});</span>
    }
<span class="cstat-no" title="statement not covered" >    this.serviceNo = text;</span>
  }
&nbsp;
  render () {
    const {buttonDisabled} = this.state;
    return (
      &lt;View style={styles.container}&gt;
        &lt;View style={styles.card}&gt;
          &lt;ISText style={styles.headerText} type='SEMIBOLD'&gt;{translate('PAYMENT_OTHERS_ENTER_PROMPT')}&lt;/ISText&gt;
          &lt;IconTextInput containerStyle={styles.inputBox} placeholder={translate('PAYMENT_OTHERS_PLACEHOLDER')} onChangeText={this.onTextInputChange}/&gt;
          &lt;Button style={styles.button} disabled={buttonDisabled} text={'Next'} onPress={this.navigate}/&gt;
        &lt;/View&gt;
      &lt;/View&gt;
    );
  }
}
PayOthersCard.propTypes = {
  fetchPaymentProducts: Proptypes.func
};
export default PayOthersCard;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Sun Dec 03 2017 16:14:06 GMT+0530 (IST)
</div>
</div>
<script src="../../../../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../../../../sorter.js"></script>
</body>
</html>
