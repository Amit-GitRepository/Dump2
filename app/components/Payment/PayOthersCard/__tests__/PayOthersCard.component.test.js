import PayOthersCard from '../PayOthersCard.component';
import React from 'react';
import renderer from 'react-test-renderer';

describe('PayOthersCard component', () => {
  it('renders correctly', () => {
    const component = renderer.create(<PayOthersCard />);
    expect(component).toMatchSnapshot();
  });
});
