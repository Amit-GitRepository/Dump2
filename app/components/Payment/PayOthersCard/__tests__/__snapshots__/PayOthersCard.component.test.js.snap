// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PayOthersCard component renders correctly 1`] = `
<View
  style={
    Object {
      "padding": 10,
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "#FFFFFF",
        "borderRadius": 8,
        "justifyContent": "space-between",
        "padding": 15,
        "shadowColor": "#303C46",
        "shadowOffset": Object {
          "height": 2,
          "width": 1,
        },
        "shadowOpacity": 0.3,
        "shadowRadius": 1,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "backgroundColor": "transparent",
          "color": "#303C46",
          "fontFamily": "SukhumvitSet-SemiBold",
          "fontSize": 16,
          "marginBottom": 10,
        }
      }
    >
      Enter Service Number
    </Text>
    <View
      accessibilityLabel={undefined}
      style={
        Array [
          Object {
            "alignSelf": "stretch",
            "backgroundColor": "#FFFFFF",
            "borderColor": "#605E71",
            "borderRadius": 4,
            "borderWidth": 2,
            "flexDirection": "row",
          },
          Object {
            "borderColor": "#B9C1C9",
            "borderRadius": 8,
            "borderWidth": 1,
            "flexDirection": "row-reverse",
            "paddingHorizontal": 5,
          },
        ]
      }
      testID={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={false}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": undefined,
              "fontSize": 20,
            },
            Object {
              "alignSelf": "center",
              "backgroundColor": "transparent",
              "color": "#B9C1C9",
              "padding": 5,
            },
            Object {
              "fontFamily": "iservice",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
          ]
        }
      >
        î¤±
      </Text>
      <TextInput
        allowFontScaling={true}
        keyboardType="numeric"
        numberOfLines={1}
        onChangeText={[Function]}
        onSubmitEditing={[Function]}
        placeholder="TrueMove H, TrueOnline, TrueVisions"
        placeholderTextColor="#B9C1C9"
        style={
          Array [
            Object {
              "alignSelf": "stretch",
              "borderRadius": 4,
              "flex": 1,
              "fontSize": 16,
              "paddingHorizontal": 0,
              "paddingVertical": 10,
            },
            Object {},
          ]
        }
        underlineColorAndroid="transparent"
      />
    </View>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      collapsable={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignSelf": "center",
          "backgroundColor": "#F93E60",
          "borderRadius": 50,
          "marginTop": 10,
          "opacity": 1,
          "padding": 10,
          "width": 100,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <View
        style={
          Object {
            "opacity": 0.7,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "alignSelf": "stretch",
              "flexDirection": "row",
              "justifyContent": "center",
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "backgroundColor": "transparent",
                  "color": "#303C46",
                },
                Array [
                  Object {
                    "color": "#FFFFFF",
                    "fontSize": 16,
                  },
                  Object {},
                ],
                Object {
                  "fontFamily": "SukhumvitSet-Bold",
                },
              ]
            }
          >
            Next
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;
